core/dbt/include/global_project/macros/materializations/common/merge.sql

```diff
@@ -18,6 +18,7 @@
 {% macro default__get_merge_sql(target, source, unique_key, dest_columns, predicates) -%}
     {%- set predicates = [] if predicates is none else [] + predicates -%}
     {%- set dest_cols_csv = get_quoted_csv(dest_columns | map(attribute="name")) -%}
+    {%- set update_columns = config.get('merge_update_columns', default = dest_columns | map(attribute="name") | list) -%}
```

Comment:
Because the column names are wrapped in `adapter.quote()` below, the values of `update_columns` need to be uppercased on Snowflake. But we wouldn't want to inadvertently uppercase a `dest_columns` value that actually _is_ mixed-case-and-quoted.

Alternatively: Should we treat the `merge_update_columns` config more like `unique_key`, with no casing/quoting automatically applied, leaving it all up to the user? That feels like it may be the better approach here.

One way to achieve that is by updating this line:
```suggestion
    {%- set update_columns = config.get('merge_update_columns', default = dest_columns | map(attribute="quoted") | list) -%}
```